segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -301,6 +303,9 @@ private void handleAppendResult(final Append append, Throwable exception, Timer
                     latestEventNumbers.put(Pair.of(append.getSegment(), append.getWriterId()), append.getEventNumber());
                 } else {
                     if (!conditionalFailed) {
+                        List<Append> appends = waitingAppends.get(append.getWriterId());
+                        pendingBytes -= appends.stream().filter(a -> !a.isConditional())
+                                .mapToInt(a -> a.getData().readableBytes()).sum();
```

Comment:
I am not sure if this will give you the results you want. If someone has read all or part of `a.getData()` then `.readableBytes()` will return less than what it was before, so your calculation will be inaccurate.

Do you have unit tests for this feature?

Reply:
I have removed this change.. because ,  this method is always called after getNextAppend .
The getNextAppend is clearing the pending bytes.
The existing unit tests covers this code diff.